src/gt4py/next/iterator/transforms/trace_shifts.py

```diff
@@ -11,23 +11,27 @@
 # distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
 #
 # SPDX-License-Identifier: GPL-3.0-or-later
-
-import types
+import enum
 from collections.abc import Callable
 from dataclasses import dataclass
-from typing import Any, Final, Union
+from typing import Any, Final, Iterable, Literal
 
 from gt4py.eve import NodeTranslator
 from gt4py.next.iterator import ir
 from gt4py.next.iterator.transforms.collect_shifts import ALL_NEIGHBORS
 
 
-VALUE_TOKEN = types.new_class("VALUE_TOKEN")
-TYPE_TOKEN = types.new_class("TYPE_TOKEN")
+class Sentinel(enum.Enum):
+    VALUE = object()
+    TYPE = object()
+
+
+class IteratorTracer:
+    pass
```

Comment:
Sorry, don't understand that point, can you elaborate?

Reply:
`abc.ABC` overrides the `isinstance` check (it's one of the core points PEP 3119: "_A way to overload isinstance() and issubclass()_") which becomes significantly slower:
```
In [2]: import abc
   ...: import timeit
   ...: 
   ...: class Foo(abc.ABC):
   ...:   pass
   ...: 
   ...: class Bar(Foo):
   ...:   pass
   ...: 
   ...: instance = Bar()
   ...: 
   ...: timeit.timeit(lambda: isinstance(instance, Foo))
Out[2]: 0.5078558220047853

In [3]: class Foo1():
   ...:   pass
   ...: 
   ...: class Bar1(Foo):
   ...:   pass
   ...: 
   ...: instance = Bar1()
   ...: 
   ...: timeit.timeit(lambda: isinstance(instance, Foo1))
Out[3]: 0.18197159600094892
```
Since I am doing these checks in many places and am worried about the performance anyway already, I would like to avoid using `abc`. Instead I would just syntactically signify what subclasses of `IteratorTracer` are expected to implement (code wise the same as `abc.ABC`, but just skipping the inheritance).

Reply:
Ok, let's keep it as is, as the abstraction is also very localized.